File: healthserver/doctor-activation-service/src/main/java/com/healthapp/doctor/dto/response/AppointmentResponse.java
package com.healthapp.doctor.dto.response;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class AppointmentResponse {

    private String id;

    // Patient Info
    private String patientId;
    private String patientEmail;
    private String patientName;
    private String patientPhone;

    // Doctor Info
    private String doctorId;
    private String doctorEmail;
    private String doctorName;
    private String specialization;

    // Appointment Details
    private LocalDateTime appointmentDateTime;
    private String appointmentType;
    private String reason;
    private String notes;
    private String status;

    // Consultation (if completed)
    private String diagnosis;
    private String prescription;
    private String doctorNotes;
    private LocalDateTime completedAt;

    private LocalDateTime createdAt;
}

File: HealthyfiTN_mobile/app/src/main/java/com/health/virtualdoctor/ui/data/models/AppointmentResponse.kt
package com.health.virtualdoctor.ui.data.models

import com.google.gson.annotations.SerializedName

data class AppointmentResponse(
    @SerializedName("id")
    val id: String,

    @SerializedName("patientId")
    val patientId: String,

    @SerializedName("patientEmail")
    val patientEmail: String,

    @SerializedName("patientName")
    val patientName: String,

    @SerializedName("patientPhone")
    val patientPhone: String?,

    @SerializedName("doctorId")
    val doctorId: String,

    @SerializedName("doctorEmail")
    val doctorEmail: String,

    @SerializedName("doctorName")
    val doctorName: String,

    @SerializedName("specialization")
    val specialization: String,

    @SerializedName("appointmentDateTime")
    val appointmentDateTime: String,

    @SerializedName("appointmentType")
    val appointmentType: String,

    @SerializedName("reason")
    val reason: String,

    @SerializedName("notes")
    val notes: String?,

    @SerializedName("status")
    val status: String,

    @SerializedName("diagnosis")
    val diagnosis: String?,

    @SerializedName("prescription")
    val prescription: String?,

    @SerializedName("doctorNotes")
    val doctorNotes: String?,

    @SerializedName("completedAt")
    val completedAt: String?,

    @SerializedName("createdAt")
    val createdAt: String
)